src/appkit.cs

```diff
@@ -17986,6 +17986,42 @@ interface NSComboBoxDelegate {
 		void SelectionIsChanging (NSNotification notification);
 	}
 
+	[BaseType (typeof(NSObject))]
+	[Model]
+	[Protocol]
+	interface NSTokenFieldCellDelegate {		
+		[Export ("tokenFieldCell:completionsForSubstring:indexOfToken:indexOfSelectedItem:")]
+		string [] GetCompletionStrings (NSTokenFieldCell tokenFieldCell, string substring, nint tokenIndex, ref nint selectedIndex);
+		
+		[Export ("tokenFieldCell:shouldAddObjects:atIndex:")]
+		NSArray ShouldAddObjects (NSTokenFieldCell tokenFieldCell, NSArray tokens, nuint index);
+		
+		[Export ("tokenFieldCell:displayStringForRepresentedObject:")]
+		string GetDisplayString (NSTokenFieldCell tokenFieldCell, NSObject representedObject);
+		
+		[Export ("tokenFieldCell:editingStringForRepresentedObject:")]
+		string GetEditingString (NSTokenFieldCell tokenFieldCell, NSObject representedObject);
+		
+		[Export ("tokenFieldCell:representedObjectForEditingString:")]
+		[return: NullAllowed]
+		NSObject GetRepresentedObject (NSTokenFieldCell tokenFieldCell, string editingString);
+		
+		[Export ("tokenFieldCell:writeRepresentedObjects:toPasteboard:")]
+		bool WriteRepresented (NSTokenFieldCell tokenFieldCell, NSArray objects, NSPasteboard pboard);
```

Comment:
for symmetry with above API a better name would be `WriteRepresentedObjects`
and the `NSArray` could be changed to `NSObject[]`